src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,49 +1,72 @@
-import React from 'react';
+import { FC, useEffect, useState } from 'react';
+import { getUser } from '../../api';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
 import { Loader } from '../Loader';
 
-export const TodoModal: React.FC = () => {
+interface Props {
+  selectedTodo: Todo | null,
+  setOpenedTodoModal: (arg0: boolean) => void,
+  loading: boolean,
+  setLoading: (arg0: boolean) => void,
+}
+
+export const TodoModal: FC<Props> = (props) => {
+  const {
+    selectedTodo, setOpenedTodoModal, loading, setLoading,
+  } = props;
+  const [selectedUser, setSelectedUser] = useState<User>();
+
+  useEffect(() => {
+    getUser(Number(selectedTodo?.userId))
+      .then(setSelectedUser);
+    setLoading(false);
+  }, [selectedTodo]);
```

Comment:
same as previous comment. 
set loading will work as soon as getUser is initialised and wil not wait until it finishes downloading data.
as a result you will ahve modal opened before data is downloaded [there should be user](https://prnt.sc/i1hcuZI07onL)